### PR TITLE
Add abs for ByteTensor and CharTensor.

### LINKED ISSUE #15089: Support torch.abs for ByteTensor, CharTensor
**UPD:** for torch.isinf I filed another issue: https://github.com/pytorch/pytorch/issues/15123

This is on `1.0.0a0+9d9e5f8`. (Can't update to 1.0 yet because of https://github.com/pytorch/pytorch/issues/14995#issuecomment-446413669)

 ```
  File ".../vadim/env/lib/python3.7/site-packages/torch/functional.py", line 288, in isinf
    return tensor.abs() == inf
RuntimeError: _abs_out is not implemented for type torch.ByteTensor
```

I think it makes sense to support `abs` on ByteTensor, even if it's a no-op. And it definitely makes sense to support `isinf` - to allow for some generic code.

### PR DESCRIPTION
**Pull request by hameerabbasi**
Fixes #15089

### CODE REVIEW

File: aten/src/ATen/Declarations.cwrap [hunk 1]
```diff
@@ -915,6 +915,8 @@
     - Long
     - Int
     - Short
+    - Byte
```
Comment (fmassa) [inline aten/src/ATen/Declarations.cwrap]: nit: can we just remove the whole `types:` part, as I think it defaults to dispatching to all types when it's not present?

File: aten/src/TH/generic/THTensorMath.h [hunk 2]
```diff
@@ -137,7 +137,7 @@ TH_API void THTensor_(eqTensorT)(THTensor *r_, THTensor *ta, THTensor *tb);
 TH_API void THTensor_(pow)(THTensor *r_, THTensor *t, scalar_t value);
 TH_API void THTensor_(tpow)(THTensor *r_, scalar_t value, THTensor *t);
 
-#if defined(TH_REAL_IS_SHORT) || defined(TH_REAL_IS_INT) || defined(TH_REAL_IS_LONG)
+#if defined(TH_REAL_IS_SHORT) || defined(TH_REAL_IS_INT) || defined(TH_REAL_IS_LONG) || defined(TH_REAL_IS_CHAR) || defined(TH_REAL_IS_BYTE)
```
Comment (fmassa) [inline aten/src/TH/generic/THTensorMath.h]: nit: Maybe now that we support all types we could instead remove the `#if` and remove the float-specific part in https://github.com/pytorch/pytorch/blob/f03296299b4f39ffed00d94c1adb12255b6d74c0/aten/src/TH/generic/THTensorMath.h#L175 ?

**Review by fmassa — COMMENTED** [review]
This looks good to me, thanks a lot!

I have a few minor comments, mostly cosmetic changes.

Also, could you add some tests in https://github.com/pytorch/pytorch/blob/f03296299b4f39ffed00d94c1adb12255b6d74c0/test/test_torch.py#L7055-L7079 ?
The cases for `uint8` and `int8` are not tested by `_test_abs` because `_make_tensors` don't take those into account https://github.com/pytorch/pytorch/blob/f03296299b4f39ffed00d94c1adb12255b6d74c0/test/test_torch.py#L112-L117

I think you can just add a few special cases there, instead of modifying `_make_tensors`.

**Comment by hameerabbasi** [comment]
That would've been my next question, where to add tests! Glad to see you already answered that.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**